scripts/tech-debt-burndown-report.ts

```diff
@@ -1,317 +0,0 @@
- * on tech debt on the current branch vs on main. The report includes some info about strictNullChecks errors,
- * as well as tracking unused (dead) code.
- */
-import fs from 'fs';
-
-//// StrictNullChecks error checking ////
-/**
- * TODO(STENCIL-446): Remove this section once `strictNullChecks` is enabled
- */
-
-/**
- * This interface is copied (a bit) from here:
- * https://github.com/aiven/tsc-output-parser/blob/master/src/parserTypes.ts
- *
- * and from manually inspecting the output of `tsc-output-parser`.
- *
- * Just so we have at least a little structure for TypeScript to grab on to
- * for the files we're `JSON.parse`-ing!
- */
-interface TSError {
-  type: 'Item';
-  value: {
-    path: {
-      type: 'Path';
-      /**
-       * The path where the error occurred
-       */
-      value: string;
-    };
-    cursor: {
-      type: 'Cursor';
-      value: {
-        line: number;
-        col: number;
-      };
-    };
-    tsError: {
-      type: 'TsError';
-      value: {
-        type: 'error';
-        /**
-         * These strings are TS error codes like `TS2339`
-         * see {@link https://github.com/microsoft/TypeScript/blob/main/src/compiler/diagnosticMessages.json}
-         */
-        errorString: string;
-      };
-    };
-  };
-}
-
-/**
- * Load JSON data, formatted by `tsc-output-parser`
- *
- * In the GH Actions workflow we write two JSON files, one for the PR
- * branch and one for main (so that we can compare them).
- *
- */
-const prData: TSError[] = JSON.parse(String(fs.readFileSync('./null_errors_pr.json')));
-const mainData: TSError[] = JSON.parse(String(fs.readFileSync('./null_errors_main.json')));
-
-const errorsOnMainCount = mainData.length;
-const errorsOnPRCount = prData.length;
-
-/**
- * Build a map which just counts the entries in an array
- *
- * @param arr the array whose entries we're going to count
- * @returns a map of counts for `arr`
- */
-const countArrayEntries = <T>(arr: Array<T>): Map<T, number> => {
-  const counts = new Map<T, number>();
-
-  arr.forEach((entry) => {
-    counts.set(entry, (counts.get(entry) ?? 0) + 1);
-  });
-
-  return counts;
-};
-
-const fileErrorCounts = countArrayEntries(prData.map((error) => error.value.path.value));
-
-const errorCodeCounts = countArrayEntries(prData.map((error) => error.value.tsError.value.errorString));
-
-//// unused exports check ////
-// TODO(STENCIL-454): Remove or change up this report once we've eliminated unused exports
-
-/**
- * ts-prune includes this string on lines in its output corresponding to items
- * which are exported and not imported anywhere but which *are* used in their
- * home modules. We want to exclude these and get only the truly dead code.
- */
-const USED_IN_MODULE = '(used in module)';
-
-/**
- * Load in data from the unused exports reports
- */
-const unusedExportsMain = String(fs.readFileSync('./unused-exports-main.txt'));
-const unusedExportsPR = String(fs.readFileSync('./unused-exports-pr.txt'));
```

Comment:
Should we remove these files as well?

Reply:
They seem to be part of the pipeline:

```
      # TODO(STENCIL-446): Remove this workflow once `strictNullChecks` is enabled
      - name: Download null errors file for main branch
        uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
        with:
          name: null_errors_main

      # TODO(STENCIL-446): Remove this workflow once `strictNullChecks` is enabled
      - name: Download null errors file for PR
        uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
        with:
          name: null_errors_pr

      # TODO(STENCIL-454): Remove or change this up once we've eliminated unused exports
      - name: Download unused exports for main
        uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
        with:
          name: unused-exports-main

      # TODO(STENCIL-454): Remove or change this up once we've eliminated unused exports
      - name: Download unused exports for PR
        uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
        with:
          name: unused-exports-pr
```

since you remove the file, we should be good!